### PR TITLE
Development updates for DFG v0.6

### LINKED ISSUE #172: Docs failing due issues in Light/Symbol/etc DFG
Hi, I'm trying to build docs in Caesar package and came across this.  Any quick ideas, or should we pick through them all and fix one by one?

cc @Affie might know best on this one?

```julia
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:249-261
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2);
│ 
│ julia> rem_edge!(g, 1, 2)
│ true
│ 
│ julia> rem_edge!(g, 1, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ using LightGraphs
│ 
│ Evaluated output:
│ 
│ ERROR: ArgumentError: Package LightGraphs not found in current path:
│ - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│ 
│ Stacktrace:
│  [1] require(::Module, ::Symbol) at ./loading.jl:876
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: ArgumentError: Package LightGraphs not found in current path:
│    - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│    
│    Stacktrace:
│     [1] require(::Module, ::Symbol) at ./loading.jl:876
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:249-261
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2);
│ 
│ julia> rem_edge!(g, 1, 2)
│ true
│ 
│ julia> rem_edge!(g, 1, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ g = SimpleGraph(2);
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: SimpleGraph not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: UndefVarError: SimpleGraph not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:249-261
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2);
│ 
│ julia> rem_edge!(g, 1, 2)
│ true
│ 
│ julia> rem_edge!(g, 1, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ add_edge!(g, 1, 2);
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: add_edge! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: UndefVarError: add_edge! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:249-261
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2);
│ 
│ julia> rem_edge!(g, 1, 2)
│ true
│ 
│ julia> rem_edge!(g, 1, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ rem_edge!(g, 1, 2)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: rem_edge! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ true
│ 
│   diff =
│    trueERROR: UndefVarError: rem_edge! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:249-261
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2);
│ 
│ julia> rem_edge!(g, 1, 2)
│ true
│ 
│ julia> rem_edge!(g, 1, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ rem_edge!(g, 1, 2)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: rem_edge! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ false
│ 
│   diff =
│    falseERROR: UndefVarError: rem_edge! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:207-217
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2)
│ true
│ 
│ julia> add_edge!(g, 2, 3)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ using LightGraphs
│ 
│ Evaluated output:
│ 
│ ERROR: ArgumentError: Package LightGraphs not found in current path:
│ - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│ 
│ Stacktrace:
│  [1] require(::Module, ::Symbol) at ./loading.jl:876
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: ArgumentError: Package LightGraphs not found in current path:
│    - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│    
│    Stacktrace:
│     [1] require(::Module, ::Symbol) at ./loading.jl:876
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:207-217
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2)
│ true
│ 
│ julia> add_edge!(g, 2, 3)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ g = SimpleGraph(2);
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: SimpleGraph not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: UndefVarError: SimpleGraph not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:207-217
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2)
│ true
│ 
│ julia> add_edge!(g, 2, 3)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ add_edge!(g, 1, 2)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: add_edge! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ true
│ 
│   diff =
│    trueERROR: UndefVarError: add_edge! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:207-217
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> add_edge!(g, 1, 2)
│ true
│ 
│ julia> add_edge!(g, 2, 3)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ add_edge!(g, 2, 3)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: add_edge! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ false
│ 
│   diff =
│    falseERROR: UndefVarError: add_edge! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:377-387
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> rem_vertex!(g, 2)
│ true
│ 
│ julia> rem_vertex!(g, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ using LightGraphs
│ 
│ Evaluated output:
│ 
│ ERROR: ArgumentError: Package LightGraphs not found in current path:
│ - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│ 
│ Stacktrace:
│  [1] require(::Module, ::Symbol) at ./loading.jl:876
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: ArgumentError: Package LightGraphs not found in current path:
│    - Run `import Pkg; Pkg.add("LightGraphs")` to install the LightGraphs package.
│    
│    Stacktrace:
│     [1] require(::Module, ::Symbol) at ./loading.jl:876
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:377-387
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> rem_vertex!(g, 2)
│ true
│ 
│ julia> rem_vertex!(g, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ g = SimpleGraph(2);
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: SimpleGraph not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ 
│ 
│   diff =
│    ERROR: UndefVarError: SimpleGraph not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:377-387
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> rem_vertex!(g, 2)
│ true
│ 
│ julia> rem_vertex!(g, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ rem_vertex!(g, 2)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: rem_vertex! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ true
│ 
│   diff =
│    trueERROR: UndefVarError: rem_vertex! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
┌ Error: doctest failure in ~/.julia/dev/DistributedFactorGraphs/src/SymbolDFG/SymbolFactorGraphs/SymbolFactorGraphs.jl:377-387
│ 
│ ```jldoctest
│ julia> using LightGraphs
│ 
│ julia> g = SimpleGraph(2);
│ 
│ julia> rem_vertex!(g, 2)
│ true
│ 
│ julia> rem_vertex!(g, 2)
│ false
│ ```
│ 
│ Subexpression:
│ 
│ rem_vertex!(g, 2)
│ 
│ Evaluated output:
│ 
│ ERROR: UndefVarError: rem_vertex! not defined
│ Stacktrace:
│  [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
│ 
│ Expected output:
│ 
│ false
│ 
│   diff =
│    falseERROR: UndefVarError: rem_vertex! not defined
│    Stacktrace:
│     [1] top-level scope at /home/dehann/.julia/dev/Caesar/docs/make.jl:15
└ @ Documenter.DocTests ~/.julia/packages/Documenter/IGqbY/src/DocTests.jl:364
```

### LINKED ISSUE #229: [FEATURE] addTags! function for post-hoc addition
This might also involve IIF somehow, but it would be great to be able to add tags through the API after a variable already exists, e.g.:
```julia
addTags!(fg, :x4, [:RADAR; :LIDAR])
```

### LINKED ISSUE #239: [BUG] TYPEDFIELDS did not work right
I had to do quick removal on master here: 34b7468  on account of fresh install error:

```julia
julia> using IncrementalInference
[ Info: Precompiling IncrementalInference [904591bb-b899-562f-9e6f-b8df64c7d480]
[ Info: Multithreaded  convolutions possible, Threads.nthreads()=1.  See `addFactor!(.;threadmodel=MultiThreaded)`.
WARNING: Method definition (::Type{DistributedFactorGraphs.VariableNodeData{T} where T<:DistributedFactorGraphs.InferenceVariable})(Array{Float64, 2}, Array{Float64, 2}, Array{Symbol, 1}, Array{Int64, 1}, Int64, Bool, Symbol, Array{Symbol, 1}, T<:DistributedFactorGraphs.InferenceVariable, Bool, Float64, Bool, Bool, Int64) where {T<:DistributedFactorGraphs.InferenceVariable} in module DistributedFactorGraphs at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/entities/DFGVariable.jl:11 overwritten at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/entities/DFGVariable.jl:41.
  ** incremental compilation may be fatally broken for this module **ERROR: LoadError: LoadError: UndefVarError: TYPEDFIELDS not defined
Stacktrace:
 [1] top-level scope at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/entities/DFGVariable.jl:123
 [2] include at ./boot.jl:328 [inlined]
 [3] include_relative(::Module, ::String) at ./loading.jl:1094
 [4] include at ./Base.jl:31 [inlined]
 [5] include(::String) at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/DistributedFactorGraphs.jl:1
 [6] top-level scope at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/DistributedFactorGraphs.jl:16
 [7] include at ./boot.jl:328 [inlined]
 [8] include_relative(::Module, ::String) at ./loading.jl:1094
 [9] include(::Module, ::String) at ./Base.jl:31
 [10] top-level scope at none:2
 [11] eval at ./boot.jl:330 [inlined]
 [12] eval(::Expr) at ./client.jl:432
 [13] top-level scope at ./none:3
in expression starting at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/entities/DFGVariable.jl:123
in expression starting at /home/tonio/.julia/packages/DistributedFactorGraphs/NMxiR/src/DistributedFactorGraphs.jl:16
ERROR: LoadError: Failed to precompile DistributedFactorGraphs [b5cc3c7e-6572-11e9-2517-99fb8daf2f04] to /home/tonio/.julia/compiled/v1.2/DistributedFactorGraphs/BWCpj.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1253
 [3] _require(::Base.PkgId) at ./loading.jl:1013
 [4] require(::Base.PkgId) at ./loading.jl:911
 [5] require(::Module, ::Symbol) at ./loading.jl:906
 [6] include at ./boot.jl:328 [inlined]
 [7] include_relative(::Module, ::String) at ./loading.jl:1094
 [8] include(::Module, ::String) at ./Base.jl:31
 [9] top-level scope at none:2
 [10] eval at ./boot.jl:330 [inlined]
 [11] eval(::Expr) at ./client.jl:432
 [12] top-level scope at ./none:3
in expression starting at /home/tonio/.julia/dev/IncrementalInference/src/IncrementalInference.jl:15
ERROR: Failed to precompile IncrementalInference [904591bb-b899-562f-9e6f-b8df64c7d480] to /home/tonio/.julia/compiled/v1.2/IncrementalInference/NrVw2.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at ./loading.jl:1253
 [3] _require(::Base.PkgId) at ./loading.jl:1013
 [4] require(::Base.PkgId) at ./loading.jl:911
 [5] require(::Module, ::Symbol) at ./loading.jl:906
```

found by @tonioteran -- thanks!

### PR DESCRIPTION
**Pull request by Affie**
partial fix for #241 
fix #239
close #229
implement #211
fix #172
part of #234

### CODE REVIEW

File: attic/MetaGraphsDFG/MetaGraphsDFG.jl [hunk 1]
```diff
@@ -18,7 +18,7 @@ export ls, lsf, getVariables, getFactors, getVariableIds, getFactorIds
 export getVariable, getFactor
 export updateVariable!, updateFactor!
 export deleteVariable!, deleteFactor!
-export getAdjacencyMatrix, getAdjacencyMatrixSparse
+export getIncidenceMatrix, getIncidenceMatrixSparse
```
Comment (Affie) [inline attic/MetaGraphsDFG/MetaGraphsDFG.jl]: To be removed completely

File: attic/MetaGraphsDFG/services/MetaGraphsDFG.jl [hunk 2]
```diff
@@ -506,7 +506,7 @@ Get an adjacency matrix for the DFG, returned as a Matrix{Union{Nothing, Symbol}
 Rows are all factors, columns are all variables, and each cell contains either nothing or the symbol of the relating factor.
 The first row and first column are factor and variable headings respectively.
 """
-function getAdjacencyMatrix(dfg::MetaGraphsDFG)::Matrix{Union{Nothing, Symbol}}
+function getIncidenceMatrix(dfg::MetaGraphsDFG)::Matrix{Union{Nothing, Symbol}}
```
Comment (Affie) [inline attic/MetaGraphsDFG/services/MetaGraphsDFG.jl]: This was moved to the attic so i missed it. Its not maintained anymore

File: attic/SymbolDFG/services/SymbolDFG.jl [hunk 3]
```diff
@@ -423,7 +423,7 @@ Get an adjacency matrix for the DFG, returned as a Matrix{Union{Nothing, Symbol}
 Rows are all factors, columns are all variables, and each cell contains either nothing or the symbol of the relating factor.
 The first row and first column are factor and variable headings respectively.
 """
-function getAdjacencyMatrix(dfg::SymbolDFG; solvable::Int=0)::Matrix{Union{Nothing, Symbol}}
+function getIncidenceMatrix(dfg::SymbolDFG; solvable::Int=0)::Matrix{Union{Nothing, Symbol}}
```
Comment (Affie) [inline attic/SymbolDFG/services/SymbolDFG.jl]: same with this, moved to attic and not maintained any more

File: attic/sandbox.jl [hunk 4]
```diff
@@ -0,0 +1,68 @@
+
+function copyGraph!(destDFG::AbstractDFG, sourceDFG::AbstractDFG, variableFactorLabels::Vector{Symbol}, includeOrphanFactors::Bool=false; copyGraphMetadata::Bool=false)
```
Comment (Affie) [inline attic/sandbox.jl]: Part of a consolidation effort on copyIntoGraphs and buildSubGraphs

File: src/Common.jl [hunk 5]
```diff
@@ -115,34 +115,37 @@ ls, lsf
...
+# end
+# function getfnctype(fact::DFGFactor, solveKey::Symbol=:default)
+#   data = getSolverData(fact) # TODO , solveKey)
+#   return getfnctype(data)
+# end
+# function getfnctype(dfg::T, lbl::Symbol, solveKey::Symbol=:default) where T <: AbstractDFG
+#   getfnctype(getFactor(dfg, exvertid))
+# end
+
+@deprecate getfnctype(args...) getFactorType(args...)
```
Comment (Affie) [inline src/Common.jl]: I just left the comments here for tracking. Can be deleted if merged and @deprecated moved.
The TODO bothered me a bit.

File: src/CommonAccessors.jl [hunk 6]
```diff
@@ -50,38 +37,35 @@ getTimestamp(v::DataLevel1) = v.timestamp
...
 
-"""
-$SIGNATURES
-
-Set the timestamp of a DFGFactor object.
-"""
-setTimestamp!(v::DataLevel1, ts::DateTime) = v.timestamp = ts
+function setTimestamp(v::DFGVariableSummary, ts::DateTime)
+    return DFGVariableSummary(v.label, ts, v.tags, v.estimateDict, v.softtypename, v.bigData, v._internalId)
+end
```
Comment (Affie) [inline src/CommonAccessors.jl]: are you happy with this behavior for `setTimestamp`, we can make a `setTimestamp!` that updates automatically under the hood

File: test/runtests.jl [hunk 7]
```diff
@@ -59,20 +63,20 @@ end
     end
 end
 
+
 if get(ENV, "IIF_TEST", "") == "true"
 
     # Switch to our upstream test branch.
-    Pkg.add(PackageSpec(name="IncrementalInference", rev="upstream/dfg_integration_test"))
+    Pkg.add(PackageSpec(name="IncrementalInference", rev="develop"))
```
Comment (Affie) [inline test/runtests.jl]: TODO change back to `upstream/dfg_integration_test` once merged

**Comment by andreasnoack** [comment]
# [Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=h1) Report
> Merging [#291](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=desc) into [master](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/commit/86cd219e37c7d079eed012ab81f3103026d3de61&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #291   +/-   ##
=======================================
  Coverage   85.32%   85.32%           
=======================================
  Files          34       34           
  Lines        1758     1758           
=======================================
  Hits         1500     1500           
  Misses        258      258           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=footer). Last update [86cd219...86cd219](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by andreasnoack** [comment]
# [Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=h1) Report
> Merging [#291](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=desc) into [master](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/commit/86cd219e37c7d079eed012ab81f3103026d3de61&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #291   +/-   ##
=======================================
  Coverage   85.32%   85.32%           
=======================================
  Files          34       34           
  Lines        1758     1758           
=======================================
  Hits         1500     1500           
  Misses        258      258           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=footer). Last update [86cd219...86cd219](https://codecov.io/gh/JuliaRobotics/DistributedFactorGraphs.jl/pull/291?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/CommonAccessors.jl (hunk 6 above)
Comment (dehann) [reply src/CommonAccessors.jl]: yes I think so, we just need to document this clearly so people who are keeping a reference to a DFGVSummary dont get a surprise that it was the wrong one for some reason (changing the timestamp in this case).  I'm assuming this is due to immutable type.  Again, we just need to document -- perhaps just in the docstring here?

File: src/LightDFG/services/LightDFG.jl [hunk 8]
```diff
@@ -72,107 +35,90 @@ function addVariable!(dfg::LightDFG{<:AbstractParams, V, <:AbstractDFGFactor}, v
...
+#     resize!(factor._variableOrderSymbols, length(variableLabels))
+#     factor._variableOrderSymbols .= variableLabels
+#
+#     @assert FactorGraphs.addFactor!(dfg.g, variableLabels, factor)
+#
+#     return factor
+# end
+
+
+function addFactor!(dfg::LightDFG{<:AbstractParams, <:AbstractDFGVariable, F}, factor::F)::F where F <: AbstractDFGFactor
```
Comment (dehann) [inline src/LightDFG/services/LightDFG.jl]: again, we should not lose the addFactor!(fg, [:x1;:x2], Pose2Pose2) usage so quickly.  We can wrap in IIF is really need to.  Also dont know how I feel about a factor being so specific list of variables if the factor is a standalone thing ... the factor is a function that requires inputs, doesn't have to be that totally specific variable in the factor graph only,

File: src/CommonAccessors.jl (hunk 6 above)
Comment (Affie) [reply src/CommonAccessors.jl]: opened issue to remind

File: src/LightDFG/services/LightDFG.jl (hunk 8 above)
Comment (Affie) [reply src/LightDFG/services/LightDFG.jl]: We will have to chat about that, if I understand you correctly what is the use of variable order symbols then?

Comment (Affie) [reply src/LightDFG/services/LightDFG.jl]: also a factor has a unique label tied to variables

**Pull request merged**